src/main/java/net/snowflake/client/jdbc/SnowflakeChunkDownloader.java

```diff
@@ -196,6 +195,10 @@ public SnowflakeChunkDownloader(SnowflakeResultSetSerializableV1 resultSetSerial
         (resultSetSerializable.getSession() != null)
             ? resultSetSerializable.getSession().orElse(null)
             : null;
+    this.memoryLimit = resultSetSerializable.getMemoryLimit();
```

Comment:
Is it possible to use `mockito` to mock this public method for testing?

Reply:
I can't think of an efficient way to do this for 2 reasons: 1. the memory limit is later changed via the void function `SnowflakeResultSetSerializableV1.adjustMemorySettings()`. and 2. If this method is mocked, it then required a mocked SnowflakeResultSetSerializableV1 object to be instantiated in order to use it. However, I want to test that the end-to-end result set creation works okay with this change, which requires calling ResultSet resultSet = statement.executeQuery(query). The creation of the ResultSetSerializableV1 object happens 4 layers of code deep from the initial ResultSet creation, so this means at least 4 other functions (but probably more) would have to be mocked as well. I think the current way of adding an extra function is the most efficient.

Reply:
Thanks for investigating on that!